dfget/core/downloader/p2p_downloader/p2p_downloader.go

```diff
@@ -207,6 +211,22 @@ func (p2p *P2PDownloader) Run() error {
 
 // Cleanup cleans all temporary resources generated by executing Run.
 func (p2p *P2PDownloader) Cleanup() {
+	if p2p.cleaned {
+		return
+	}
+
+	if !stringutils.IsEmptyStr(p2p.targetFile) {
+		fileutils.DeleteFile(p2p.targetFile)
+	}
+
+	if !stringutils.IsEmptyStr(p2p.clientFilePath) {
+		fileutils.DeleteFile(p2p.clientFilePath)
+	}
+
+	if !stringutils.IsEmptyStr(p2p.serviceFilePath) {
```

Comment:
It's better to notify the `dfget server` if it's existing to do some cleanup works.